rewrite-java/src/main/java/org/openrewrite/java/MethodMatcher.java

```diff
@@ -322,12 +322,14 @@ private AspectjUtils() {
      * An embedded * in an identifier matches any sequence of characters, but
      * does not match the package (or inner-type) separator ".".
      * <p>
+     * A full wildcard "*" will match any sequences of characters included package separators.
+     * <p>
      * The ".." wildcard matches any sequence of characters that start and end with a ".", so it can be used to pick out all
      * types in any subpackage, or all inner types. e.g. <code>within(com.xerox..*)</code> picks out all join points where
      * the code is in any declaration of a type whose name begins with "com.xerox.".
      */
     public static String aspectjNameToPattern(String name) {
-        return name
+        return name.equals("*") ? "[^.].*" : name
```

Comment:
Regex ([a-zA-Z_$][a-zA-Z\d_$]*\.)*[a-zA-Z_$][a-zA-Z\d_$]*
may be considered more accurate, but I kept the pattern similar to what exists already.